### PR TITLE
librbd: fix discard granularity for pwl cache

### PR DESCRIPTION
**Pull request by majianpeng**
## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [x] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by majianpeng** [comment]
@ideepika  please review.

File: src/librbd/cache/pwl/InitRequest.cc [hunk 1]
```diff
@@ -178,12 +178,6 @@ void InitRequest<I>::handle_set_feature_bit(int r) {
     shutdown_image_cache();
   }
 
-  if (m_image_ctx.discard_granularity_bytes) {
```
Comment (ideepika) [inline src/librbd/cache/pwl/InitRequest.cc:181]: @majianpeng is it true to assume that we would not need discard_granualarity_bytes reset when an existing image is init with cache?

**Review by ideepika — CHANGES_REQUESTED** [review]
nit added

File: src/librbd/cache/pwl/InitRequest.cc (hunk 1 above)
Comment (majianpeng) [reply src/librbd/cache/pwl/InitRequest.cc:181]: Yes, apply_metadata call two times. One time before init and other time after init. So in init function reset don't affect.

**Comment by majianpeng** [comment]
retest this please

**Review by ideepika — APPROVED** [review]
thanks @majianpeng

**Review by trociny — APPROVED** [review]
LGTM

**Comment by sunnyku** [comment]
jenkins retest this please

**Review by sunnyku — APPROVED** [review]
LGTM, thanks

**Comment by ideepika** [comment]
https://pulpito.ceph.com/ideepika-2021-12-02_12:23:25-rbd-wip-deepika-testing-2021-12-02-1334-distro-default-smithi/
https://tracker.ceph.com/issues/53112 [nfs mount issue]
[.MaxWhileTries: reached maximum tries (90) after waiting for 540 seconds]
https://tracker.ceph.com/issues/53138 [pg degrade]
https://tracker.ceph.com/issues/53477 [rbd mirror daemon crash]

**Pull request merged**